src/couch/src/couch_db.erl

```diff
@@ -200,7 +200,12 @@ reopen(#db{main_pid = Pid, fd = Fd, fd_monitor = OldRef, user_ctx = UserCtx}) ->
 
 incref(#db{fd = Fd} = Db) ->
     Ref = erlang:monitor(process, Fd),
-    {ok, Db#db{fd_monitor = Ref}}.
+    receive
+        {'DOWN', Ref, _, _, _} ->
```

Comment:
You should pattern match the reason element as noproc so we're asserting that the process was dead before we tried to acquire the monitor. I'd also swap the return value to be {ok, Db} on success, or retry on failure. No need to return the input and pattern match it on use.

Reply:
Thinking more on this, I'd use the atom down instead of retry. Was thinking in terms of the couch_server's response, but that's not appropriate when considering incref more abstractly. So something like `-spec incref(Db::db()) -> {ok, NewDb::db()} | down.`